CRM/MembershipExtras/Form/PaymentScheme.php

```diff
@@ -151,8 +164,54 @@ private function addPaymentProcessorField() {
     }
     $select = ['' => ts('- select -')] + $paymentProcessors;
 
-    $this->add('select', 'payment_processor', E::ts('Payment Processor'), $select, TRUE);
+    $this->add('select', 'payment_processor', ts('Payment Processor'), $select, TRUE);
 
   }
 
+  private function isDeleteContext() {
+    return $this->_action == CRM_Core_Action::DELETE;
+  }
+
+  private function postProcessDeletion() {
```

Comment:
The logic in this method  belongs to the PaymentScheme BAO class not to the form so it would be enforced on all calls to delete a  Payment Scheme.